osu.Game.Rulesets.Osu/Mods/OsuModBubbles.cs

```diff
@@ -0,0 +1,255 @@
+            currentCombo.BindValueChanged(combo =>
+                maxSize = Math.Min(1.75f, (float)(1.25 + 0.005 * combo.NewValue)), true);
+
+            scoreProcessor.NewJudgement += result =>
+            {
+                if (result.HitObject is not OsuHitObject osuHitObject) return;
+
+                DrawableOsuHitObject drawableOsuHitObject = lastJudgedHitobject;
+
+                switch (result.HitObject)
+                {
+                    case Slider:
+                    case SpinnerTick:
+                        break;
+
+                    default:
+                        addBubble();
+                        break;
+                }
+
+                void addBubble()
+                {
+                    BubbleDrawable bubble = bubblePool.Get();
+                    bubble.Info = new BubbleInfo
+                    {
+                        InitialSize = new Vector2(bubbleRadius),
+                        MaxSize = maxSize,
+                        Position = getPosition(),
+                        FadeTime = bubbleFade,
+                        Colour = drawableOsuHitObject.AccentColour.Value,
+                        IsHit = drawableOsuHitObject.IsHit,
+                    };
+                    bubbleContainer.Add(bubble);
+                }
+
+                Vector2 getPosition()
+                {
+                    switch (drawableOsuHitObject)
+                    {
+                        // SliderHeads are derived from HitCircles,
+                        // so we must handle them before to avoid them using the wrong positioning logic
+                        case DrawableSliderHead:
+                            return osuHitObject.Position;
+
+                        // Using hitobject position will cause issues with HitCircle placement due to stack leniency.
+                        case DrawableHitCircle:
+                            return drawableOsuHitObject.Position;
+
+                        default:
+                            return osuHitObject.Position;
+                    }
+                }
+            };
+
+            scoreProcessor.JudgementReverted += _ =>
+            {
+                bubbleContainer.LastOrDefault()?.FinishTransforms();
+                bubbleContainer.LastOrDefault()?.Expire();
+            };
+        }
+
+        public void ApplyToDrawableRuleset(DrawableRuleset<OsuHitObject> drawableRuleset)
+        {
+            // Multiplying by 2 results in an initial size that is too large, hence 1.90 has been chosen
+            // Also avoids the HitObject bleeding around the edges of the bubble drawable at minimum size
+            bubbleRadius = (float)(drawableRuleset.Beatmap.HitObjects.OfType<HitCircle>().First().Radius * 1.90f);
+            bubbleFade = drawableRuleset.Beatmap.HitObjects.OfType<HitCircle>().First().TimePreempt * 2;
+
+            // We want to hide the judgements since they are obscured by the BubbleDrawable (due to layering)
```

Comment:
Obscured in all cases, or just on the skin you tried?

Reply:
Well, ive tried on argon, triangles, classic, and a bunch of customs, and only triangles really looks ok, because the text gets hidden but the triangles surround the "bubble". this was moreso to avoid text poking out from under bubbles on any skin that might have ridiculously large  judgements, but hey im open to changing it, moreso i thought this was a fine stop gap and that in the future they could perhaps be proxied above the bubbles somehow